Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/jobs/JobUtils.java

```diff
@@ -410,8 +415,13 @@ public static String getInformationFromYarnLogDom(final CredentialsProvider cred
     }
 
     public static HttpEntity getEntity(@NotNull final IClusterDetail clusterDetail, @NotNull final String url) throws IOException, HDIException {
-        provider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(clusterDetail.getHttpUserName(), clusterDetail.getHttpPassword()));
-        final HttpClient client = HttpClients.custom()
+        if (StringUtils.isNoneEmpty(clusterDetail.getHttpUserName()) && StringUtils.isNoneEmpty(clusterDetail.getHttpPassword())) {
+            provider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(clusterDetail.getHttpUserName(), clusterDetail.getHttpPassword()));
+        }
+
+        final HttpClient client = clusterDetail instanceof MfaEspCluster
```

Comment:
Could you expend the MfsEspcluster check as an `if..else..` with combination the above codes?